src/test/create-user.test.ts

```diff
@@ -1,116 +1,114 @@
-import { CryptoService } from '@core';
-import { Database, UserDataSource } from '@db';
-import { CreateUserInputModel, CreateUserUseCase, UserModel } from '@domain';
-import { User } from '@entities';
+import { HOST, PORT } from '@core';
+import { CreateUserInputModel, CreateUserResponseModel } from '@domain';
+import axios, { AxiosResponse } from 'axios';
 import { expect } from 'chai';
-import { before, describe } from 'mocha';
-import { Container, Service } from 'typedi';
+import { afterEach, describe } from 'mocha';
+import { Container } from 'typedi';
+import { clearRepository } from './index';
 
-describe('GraphQL => Unit Test - Create User use-case', () => {
-  let saltTest: string;
-  let repository: UserDataSourceTest;
-  let createUser: CreateUserUseCaseTest;
-
-  class CreateUserUseCaseTest extends CreateUserUseCase {
-    constructor() {
-      super((repository = Container.get(UserDataSourceTest)), Container.get(CryptoService));
-    }
-    saltTest: string = this.cryptoService.generateSalt();
-
-    async execTest(input: CreateUserInputModel): Promise<UserModel> {
-      const { name, email, birthdate, password } = input;
-      return this.repository.saveUser({
-        name,
-        birthdate: new Date(birthdate).toJSON(),
-        email,
-        password,
-        salt: saltTest,
-      });
+describe('Axios => Integration Testing - Create User use-case', () => {
+  class GraphQLEndPoint {
+    static getURL(): string {
+      const host: string = Container.get(HOST);
+      const port: number = Container.get(PORT);
+      return `http://${host}:${port}/`;
     }
   }
 
-  @Service()
-  class UserDataSourceTest extends UserDataSource {
-    override async saveUser(user: CreateUserInputModel) {
-      Object.assign(user, { salt: saltTest });
-      return this.userRepository.save({ ...user });
-    }
+  type GraphRespModel = {
+    createUser: CreateUserResponseModel;
+  };
+  type AxiosResp = Pick<AxiosResponse, 'data' | 'status' | 'statusText'> & { data: { errors?: any[] } };
+
+  let graphResponse: GraphRespModel & { errors: any[] };
+  let input = (user?: Partial<CreateUserInputModel>) => {
+    return `#graphql
+    mutation {
+      createUser (input: { 
+        name: "John",
+        email: "${user?.email ?? 'jo@example.com'}",
+        birthdate: "${user?.birthdate ?? '2000-01-01'}",
+        password: "${user?.password ?? 'p@ssw0rd'}"
+      }) {
+        id
+        name
+        email
+        birthdate
+      }
+    }`;
+  };
+
+  async function requisition(queryInput: string): Promise<AxiosResp> {
+    return axios.post(GraphQLEndPoint.getURL(), { query: queryInput });
   }
```

Comment:
tem um jeito mais simples de fazer isso:

```suggestion
  const defaultUserInput = {
    name: "John",
    email: "jo@example.com",
    birthdate: "2000-01-01",
    password: "p@ssw0rd",
  };
  const query = `
    mutation($input: CreateUserInput!) {
      createUser (input: $input) {
        id
        name
        email
        birthdate
      }
    }
  `;

  async function requisition(queryInput: string, user?: Partial<CreateUserInputModel>): Promise<AxiosResp> {
    const variables = Object.assing(defaultUserInput, user);
    return axios.post(GraphQLEndPoint.getURL(), { data: { query, variables } });
  }
```

Desta forma, a sua funcao `requisition` recebe apenas objetos como input.

Reply:
não funciona dessa forma